src/unix/fs.c

```diff
@@ -1180,7 +1181,11 @@ static ssize_t uv__fs_copyfile(uv_fs_t* req) {
   bytes_to_send = src_statsbuf.st_size;
   in_offset = 0;
   while (bytes_to_send != 0) {
-    uv_fs_sendfile(NULL, &fs_req, dstfd, srcfd, in_offset, bytes_to_send, NULL);
+    bytes_chunk = SIZE_MAX >> 1; // highest value for size_t
```

Comment:
I think INT32_MAX may be a better choice, as libuv otherwise sometimes get confused (cf. https://github.com/libuv/libuv/pull/1501)

Reply:
but do you have any guarantee that `size_t` and `int32_t` will be the same number of bytes everywhere? If there was a platform where for some reason `size_t` was only 2 bytes, it would overflow.

Reply:
I don't think it's worth wasting time considering hypothetical platforms that aren't supported anyways. But yes, if someone later needed to support for such a platform, they'd instead here need to use the smaller of the two values. (Possibly, they might want to use `off_t` here instead too, but note that the maximum it'll ever transfer on linux is hardcoded in the kernel to 0x7ffff000)

Reply:
I think that the way I wrote it is 100% safe. Sure it's a bit more ugly to do it like this with the bit shift but it's just gonna work everywhere for sure. Also, the issue that you shared, that it might throw errors if you try to write more than 2GB at a time is not a problem, because `SIZE_MAX >> 1` is the same as `INT32_MAX`, they're both `0x7fffffff`.

But if you insist, I can change it to `INT32_MAX` I guess.

Reply:
```suggestion
    bytes_chunk = SSIZE_MAX;
```
I believe that is what you're after? The logic looks otherwise correct to me.

The thing @vtjnash links to is handled by `uv_fs_sendfile()`, it falls back to a - sadly not very efficient - read/write loop when `sendfile(2)` fails with EINVAL.

Can you drop the braces around the `if` statement?

Reply:
yeah actually I didn't know about `SSIZE_MAX`, that's perfect, thanks!

Ok, I'll drop the braces.